### PR TITLE
Fix #65 - Skip unknown batch ids

### LINKED ISSUE #65: Set of valid batch ids should be filtered in staging
https://github.com/mozilla/prio-processor/blob/3cdc368707f8dc0f917d7b3d537c31645f4260f7/processor/prio_processor/staging.py#L77-L88

The partitioning job should limit the set of valid batch ids to those set in `content.json`. 

```
[2019-08-07 23:44:33,104] {logging_mixin.py:95} INFO - [2019-08-07 23:44:33,103] {pod_launcher.py:104} INFO - + parallel process ::: 'submission_date=2019-08-05/batch_id=content.blocking_blocked_TESTONLY-0/part-00000-389a770c-9507-4c6e-a669-160366928daf.c000.json
[2019-08-07 23:44:33,105] {logging_mixin.py:95} INFO - [2019-08-07 23:44:33,105] {pod_launcher.py:104} INFO - submission_date=2019-08-05/batch_id=content.blocking_blocked-1/part-00000-389a770c-9507-4c6e-a669-160366928daf.c000.json
[2019-08-07 23:44:33,105] {logging_mixin.py:95} INFO - [2019-08-07 23:44:33,105] {pod_launcher.py:104} INFO - submission_date=2019-08-05/batch_id=content.blocking_blocked-0/part-00000-389a770c-9507-4c6e-a669-160366928daf.c000.json
[2019-08-07 23:44:33,106] {logging_mixin.py:95} INFO - [2019-08-07 23:44:33,106] {pod_launcher.py:104} INFO - submission_date=2019-08-05/batch_id=content.blocking_blocked_TESTONLY-1/part-00000-389a770c-9507-4c6e-a669-160366928daf.c000.json'
[2019-08-07 23:44:42,679] {logging_mixin.py:95} INFO - [2019-08-07 23:44:42,678] {pod_launcher.py:104} INFO - Running verify1
[2019-08-07 23:44:42,681] {logging_mixin.py:95} INFO - [2019-08-07 23:44:42,681] {pod_launcher.py:104} INFO - Traceback (most recent call last):
[2019-08-07 23:44:42,684] {logging_mixin.py:95} INFO - [2019-08-07 23:44:42,683] {pod_launcher.py:104} INFO -   File "/usr/local/bin/prio", line 11, in <module>
[2019-08-07 23:44:42,688] {logging_mixin.py:95} INFO - [2019-08-07 23:44:42,687] {pod_launcher.py:104} INFO -     sys.exit(main())
[2019-08-07 23:44:42,690] {logging_mixin.py:95} INFO - [2019-08-07 23:44:42,689] {pod_launcher.py:104} INFO -   File "/usr/local/lib64/python3.6/site-packages/click/core.py", line 764, in __call__
[2019-08-07 23:44:42,691] {logging_mixin.py:95} INFO - [2019-08-07 23:44:42,691] {pod_launcher.py:104} INFO -     return self.main(*args, **kwargs)
[2019-08-07 23:44:42,693] {logging_mixin.py:95} INFO - [2019-08-07 23:44:42,692] {pod_launcher.py:104} INFO -   File "/usr/local/lib64/python3.6/site-packages/click/core.py", line 717, in main
[2019-08-07 23:44:42,695] {logging_mixin.py:95} INFO - [2019-08-07 23:44:42,694] {pod_launcher.py:104} INFO -     rv = self.invoke(ctx)
[2019-08-07 23:44:42,697] {logging_mixin.py:95} INFO - [2019-08-07 23:44:42,696] {pod_launcher.py:104} INFO -   File "/usr/local/lib64/python3.6/site-packages/click/core.py", line 1137, in invoke
[2019-08-07 23:44:42,699] {logging_mixin.py:95} INFO - [2019-08-07 23:44:42,698] {pod_launcher.py:104} INFO -     return _process_result(sub_ctx.command.invoke(sub_ctx))
[2019-08-07 23:44:42,700] {logging_mixin.py:95} INFO - [2019-08-07 23:44:42,700] {pod_launcher.py:104} INFO -   File "/usr/local/lib64/python3.6/site-packages/click/core.py", line 956, in invoke
[2019-08-07 23:44:42,701] {logging_mixin.py:95} INFO - [2019-08-07 23:44:42,701] {pod_launcher.py:104} INFO -     return ctx.invoke(self.callback, **ctx.params)
[2019-08-07 23:44:42,704] {logging_mixin.py:95} INFO - [2019-08-07 23:44:42,703] {pod_launcher.py:104} INFO -   File "/usr/local/lib64/python3.6/site-packages/click/core.py", line 555, in invoke
[2019-08-07 23:44:42,707] {logging_mixin.py:95} INFO - [2019-08-07 23:44:42,706] {pod_launcher.py:104} INFO -     return callback(*args, **kwargs)
[2019-08-07 23:44:42,709] {logging_mixin.py:95} INFO - [2019-08-07 23:44:42,709] {pod_launcher.py:104} INFO -   File "/usr/local/lib64/python3.6/site-packages/prio/cli/commands.py", line 134, in verify1
[2019-08-07 23:44:42,711] {logging_mixin.py:95} INFO - [2019-08-07 23:44:42,711] {pod_launcher.py:104} INFO -     libprio.PrioVerifier_set_data(verifier, share)
[2019-08-07 23:44:42,714] {logging_mixin.py:95} INFO - [2019-08-07 23:44:42,713] {pod_launcher.py:104} INFO -   File "/usr/local/lib64/python3.6/site-packages/prio/libprio.py", line 319, in PrioVerifier_set_data
[2019-08-07 23:44:42,716] {logging_mixin.py:95} INFO - [2019-08-07 23:44:42,715] {pod_launcher.py:104} INFO -     return _libprio.PrioVerifier_set_data(*args)
[2019-08-07 23:44:42,717] {logging_mixin.py:95} INFO - [2019-08-07 23:44:42,717] {pod_launcher.py:104} INFO - RuntimeError: PrioVerifier_set_data was not successful.
[2019-08-07 23:44:42,846] {logging_mixin.py:95} INFO - [2019-08-07 23:44:42,845] {pod_launcher.py:104} INFO - Usage: prio verify1 [OPTIONS]
[2019-08-07 23:44:42,847] {logging_mixin.py:95} INFO - [2019-08-07 23:44:42,846] {pod_launcher.py:104} INFO - Try "prio verify1 --help" for help.
[2019-08-07 23:44:42,848] {logging_mixin.py:95} INFO - [2019-08-07 23:44:42,848] {pod_launcher.py:104} INFO -
[2019-08-07 23:44:42,849] {logging_mixin.py:95} INFO - [2019-08-07 23:44:42,849] {pod_launcher.py:104} INFO - Error: Invalid value for "--n-data": null is not a valid integer
[2019-08-07 23:44:43,062] {logging_mixin.py:95} INFO - [2019-08-07 23:44:43,062] {pod_launcher.py:104} INFO - Usage: prio verify1 [OPTIONS]
[2019-08-07 23:44:43,063] {logging_mixin.py:95} INFO - [2019-08-07 23:44:43,063] {pod_launcher.py:104} INFO - Try "prio verify1 --help" for help.
[2019-08-07 23:44:43,064] {logging_mixin.py:95} INFO - [2019-08-07 23:44:43,064] {pod_launcher.py:104} INFO -
[2019-08-07 23:44:43,065] {logging_mixin.py:95} INFO - [2019-08-07 23:44:43,065] {pod_launcher.py:104} INFO - Error: Invalid value for "--n-data": null is not a valid integer
```

### PR DESCRIPTION
**Pull request by acmiyaguchi**
This fixes issues describe in #65.

### CODE REVIEW

File: processor/tests/test_bin_process.py [hunk 1]
```diff
@@ -226,3 +226,16 @@ def _validate(fs, server_env):
 
     _validate(gcsfs_a(), server_a_env)
     _validate(gcsfs_b(), server_b_env)
+
+    # assert that bad-id is not in the final published set
+    paths = gcsfs_a().walk(server_a_env["BUCKET_INTERNAL_PRIVATE"])
+    processed = [p for p in paths if "processed" in p]
+    raw = [p for p in paths if "raw" in p]
+
+    # query contains bad-id and other valid ids
```
Comment (wlach) [inline processor/tests/test_bin_process.py]: should this be "raw query"?

File: processor/tests/test_bin_process.py [hunk 2]
```diff
@@ -226,3 +226,16 @@ def _validate(fs, server_env):
...
+    assert any(["batch_id=bad-id" in p for p in raw])
+    assert any(["batch_id=content.blocking_blocked_TESTONLY-0" in p for p in raw])
+
+    # query does not contain bad-id, since it's skipped
```
Comment (wlach) [inline processor/tests/test_bin_process.py]: Likewise, should this be "processed query"?

File: processor/bin/process [hunk 3]
```diff
@@ -222,6 +226,10 @@ function verify2() {
     function process() {
         local filename=$1
         export_data_dimensions "$filename"
+        if [[ $N_DATA == "null" ]]; then
+            echo "$BATCH_ID has unknown dimension, skipping..."
```
Comment (wlach) [inline processor/bin/process]: Are these scripts run in production? Hopefully this won't generate too much output?

Comment (acmiyaguchi) [reply processor/bin/process]: Yes, this gets run in production. It shouldn't be too noisy (the number of lines is equal to the number of partitions), and should only occur during `verify1`.

File: processor/tests/test_bin_process.py (hunk 1 above)
Comment (acmiyaguchi) [reply processor/tests/test_bin_process.py]: I was using query in the sense that the following assertions are queries on the path list e.g. "assert any path in raw contains `batch_id=bad-id`". I reworded the comments so they align with the variables a bit more.

**Pull request merged**